cartographer/mapping/internal/2d/pose_graph_2d.cc

```diff
@@ -480,6 +482,25 @@ void PoseGraph2D::HandleWorkQueue(
     num_nodes_since_last_loop_closure_ = 0;
   }
 
+  // Update the gauges that count the current number of constraints.
```

Comment:
Could be an extra function (I leave this to you)

Reply:
As long as it's in `HandleWorkQueue` I don't see the need to call it somewhere else.